presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -204,6 +205,33 @@ public static long fromISO8601Timestamp(ConnectorSession session, @SqlType(Stand
         return packDateTimeWithZone(parseDateTimeHelper(formatter, iso8601DateTime.toStringUtf8()));
     }
 
+    @ScalarFunction("date")
+    @SqlType(StandardTypes.DATE)
+    public static long dateFromVarchar(ConnectorSession session, @SqlType(StandardTypes.VARCHAR) Slice date)
```

Comment:
It turns out you can achieve the same result by annotating the function that implements the cast from varchar->date with `@ScalarFunction("date")` (I checked to be sure). Take a look at `DateOperators.castFromSlice`, `TimestampOperators.castToDate` and `TimestampWithTimezoneOperators.castToDate`.